cylc/flow/data_store_mgr.py

```diff
@@ -295,6 +295,17 @@ def runtime_from_config(rtconfig):
     )
 
 
+def reset_protobuf_object(msg_class, msg_orig):
+    """Reset object to clear memory build-up."""
+    # See: https://github.com/protocolbuffers/protobuf/issues/19674
+    # The new message instantiation needs happen on a separate line.
+    new_msg = msg_class()
+    new_msg.ParseFromString(
+        msg_orig.SerializeToString(deterministic=True)
+    )
```

Comment:
Ouch, we have to serialise and deserialise the whole workflow object every time something change in it?

That must be a performance hit?

Reply:
Well, I had to weigh up doing this instead of using the python implementation (i.e. `export PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python`):
- the underlying `upb` libraries are faster than `python` implementation at doing this..
- Also it's not a large object, and only one.
- when I timed it, it was ~0.0001sec before and after this change

Reply:
Essentially what it's doing is copying the content of the old message into a new message.. I'll try some alternatives, cause it may be enough just to create a new object and copy from the old...  One moment

(I just did it this way to ensure I wasn't transferring memory references from the old object)

Reply:
Well, what do you know.. It worked with existing tools (`CopyFrom`), changed to that.

Reply:
Trying to work out whether it would make more sense to run this as a periodic activity (with a set interval), rather than doing it for every update (which could potentially be every main loop cycle):

E.g, like the health check routine: https://github.com/cylc/cylc-flow/blob/master/cylc/flow/main_loop/health_check.py

Reply:
I think it's not that expensive to run on `PbWorkflow` alone .. Especially since we've swapped to `CopyFrom`, also, this solution will be adopted by the UIS without the need to implement the same here